gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/filter/SSOCookieFederationFilter.java

```diff
@@ -146,11 +146,11 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
     if (ssoCookies.isEmpty()) {
       /* check for unauthenticated paths to bypass */
       for (final String path : unAuthenticatedPaths) {
-        if (req.getRequestURI().contains(path)) {
+        if (req.getPathInfo().equals(path)) {
           /* This path is configured as an unauthenticated path let the request through */
           final Subject sub = new Subject();
           sub.getPrincipals().add(new PrimaryPrincipal("anonymous"));
-          LOGGER.unauthenticatedPathBypass(path, req.getRequestURI());
+          LOGGER.unauthenticatedPathBypass(req.getRequestURI(), unAuthenticatedPaths.toString());
           continueWithEstablishedSecurityContext(sub, req, res, chain);
         }
```

Comment:
Is a break statement needed here to terminate the for loop?

Reply:
Refactored this code so this should not be an issue.